client/src/main/java/io/pravega/client/byteStream/impl/ByteStreamWriterImpl.java

```diff
@@ -32,6 +35,7 @@
     private final SegmentOutputStream out;
     @NonNull
     private final SegmentMetadataClient meta;
+    private CompletableFuture<Void> latestEventFuture;
```

Comment:
A "guardedby" annotation should be on each mutable field to indicate which lock protects it.